### PR TITLE
Add support for loongarch64

### PR DESCRIPTION
**Pull request by haomi123**
Add support for loongarch64

### CODE REVIEW

File: src/arch/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@ elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "s390" OR
     add_library(arch s390.cpp)
 elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
     add_library(arch x86_64.cpp)
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "loongarch32" OR
+       CMAKE_SYSTEM_PROCESSOR STREQUAL "loongarch64")
+    add_library(arch loongarch64.cpp)
```
Comment (fbs) [inline src/arch/CMakeLists.txt:15]: is this correct for 32?

File: src/arch/loongarch64.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+#include <array>
+
+// SP points to the first argument that is passed on the stack
+#define ARG0_STACK 0
+
+namespace bpftrace {
+namespace arch {
+
+// clang-format off
+static std::array<std::string, 17> registers = {
```
Comment (chenhengqi) [inline src/arch/loongarch64.cpp:13]: I have an unfinished branch at https://github.com/iovisor/bpftrace/compare/master...chenhengqi:bpftrace:support-loongarch

Did you really test this change ?

File: src/arch/CMakeLists.txt (hunk 1 above)
Comment (haomi123) [reply src/arch/CMakeLists.txt:15]: yes.  it is correct

File: src/arch/loongarch64.cpp (hunk 2 above)
Comment (haomi123) [reply src/arch/loongarch64.cpp:13]: I've done compile verification locally

Comment (haomi123) [reply src/arch/loongarch64.cpp:13]: > I have an unfinished branch at [master...chenhengqi:bpftrace:support-loongarch](https://github.com/iovisor/bpftrace/compare/master...chenhengqi:bpftrace:support-loongarch)
> 
> Did you really test this change ?

I've done compile verification locally

Comment (chenhengqi) [reply src/arch/loongarch64.cpp:13]: Just opened #2466, your change does NOT work actually.

**Comment by haomi123** [comment]
@fbs @viktormalik  @ajor  @danobi @chenhengqi @BurntBrunch   please review and merge, thank you!

File: src/arch/loongarch64.cpp (hunk 2 above)
Comment (haomi123) [reply src/arch/loongarch64.cpp:13]: > Just opened #2466, your change does NOT work actually.

Why didn't my changes actually work

Comment (viktormalik) [reply src/arch/loongarch64.cpp:13]: Your implementation does not comply with the [ABI specification](https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html). For example, it says that there are 32 general-purpose registers while you only state 17, etc.

#2466 looks much better on that front, so I'd favor that one over this PR.

**Comment by viktormalik** [comment]
Support for loongarch is now merged (#2466).

**Pull request closed without merging**